next/test/visibility.test.ts

```diff
@@ -1,271 +1,267 @@
-                hasJob: { const: true },
+        },
+      },
+      if: {
+        properties: {
+          form: {
+            properties: {
+              name: {
+                const: 'admin',
               },
-              required: ['hasJob'],
             },
-            then: {
-              required: ['salary'],
+            required: ['name'],
+          },
+        },
+        required: ['form'],
+      },
+      else: {
+        properties: {
+          form: {
+            properties: {
+              password: false,
             },
           },
-        ],
-      }
-
-      const form = createHeadlessForm(schema)
-
-      const hasJobField = form.fields.find(f => f.name === 'hasJob')
-      const salaryField = form.fields.find(f => f.name === 'salary')
-
-      expect(hasJobField?.isVisible).toBe(false)
-      expect(salaryField?.isVisible).toBe(false)
-
-      const value1: ObjectValue = { isAdult: true as unknown as SchemaValue }
-      form.handleValidation(value1)
-      expect(hasJobField?.isVisible).toBe(true)
-      expect(salaryField?.isVisible).toBe(false)
-
-      const value2: ObjectValue = {
-        isAdult: true as unknown as SchemaValue,
-        hasJob: true as unknown as SchemaValue,
-      }
-      form.handleValidation(value2)
-      expect(hasJobField?.isVisible).toBe(true)
-      expect(salaryField?.isVisible).toBe(true)
-
-      const value3: ObjectValue = {
-        isAdult: true as unknown as SchemaValue,
-        hasJob: false as unknown as SchemaValue,
-      }
-      form.handleValidation(value3)
-      expect(hasJobField?.isVisible).toBe(true)
-      expect(salaryField?.isVisible).toBe(false)
+        },
+      },
+    }
+
+    it('should hide the password field by default', () => {
+      const form = createHeadlessForm(schema, { initialValues: { form: { name: '', password: null } } })
+      // No name provided
+      expect(form.fields.find(field => field.name === 'form')?.fields?.find(field => field.name === 'password')?.isVisible).toBe(false)
+
+      // Different name provided
+      form.handleValidation({
+        form: {
+          name: 'some name',
+          password: null,
+        },
+      })
+      expect(form.fields.find(field => field.name === 'form')?.fields?.find(field => field.name === 'password')?.isVisible).toBe(false)
+    })
 
-      const value4: ObjectValue = {
-        isAdult: false as unknown as SchemaValue,
-      }
-      form.handleValidation(value4)
-      expect(hasJobField?.isVisible).toBe(false)
-      expect(salaryField?.isVisible).toBe(false)
+    it('should show the password field if the name is admin', () => {
+      const form = createHeadlessForm(schema, { initialValues: { form: { name: 'admin', password: null } } })
+      form.handleValidation({ form: {
+        name: 'admin',
+      } })
+      expect(form.fields.find(field => field.name === 'form')?.fields?.find(field => field.name === 'password')?.isVisible).toBe(true)
     })
   })
 })
```

Comment:
bugs: I tried it against other demos in the JSF docs, I noticed these scenarios don't work yet:

- Hidden conditional read-only field
- Conditional properties
- Field with conditional statement
- Dynamic properties - select field

Probably this will be other PRs right?

Reply:
Yes, this PR is only for visibility